qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -555,22 +555,17 @@ def _is_target_series(datum, **filters):
                 f"X value key {x_key} is not defined in circuit metadata."
             ) from ex
 
-        y_values, y_sigmas = zip(*map(data_processor, data))
-
-        # TODO this should be handled in data processor.
-        # Future data processor may take full sequence of data rather than datum.
-        # The CurveAnalysis can pass series filter_kwargs to the processor
-        # so that it can filter data to extract.
-        if self._get_option("normalization"):
-            y_min, y_max = min(y_values), max(y_values)
-            scale = 1 / (y_max - y_min)
+        if isinstance(data_processor, DataProcessor) and data_processor.requires_all_data():
+            y_values, y_sigmas = data_processor(data)
+            if y_sigmas is None:
+                y_sigmas = np.full(y_values.shape, None)
         else:
-            scale = 1.0
+            y_values, y_sigmas = zip(*map(data_processor, data))
```

Comment:
This should be handled by data processor. We can always pass data to process as a list of it.

Reply:
Perhaps for a follow up PR.